R/checkSummations.R

```diff
@@ -105,12 +112,26 @@ checkSummations <- function(mifFile, outputDirectory = ".", template = NULL, sum
     # calculate differences for comparison
     comp <- comp %>%
       group_by(!!!syms(c("model", "scenario", "region", "period", "variable", "unit", "value"))) %>%
-      summarise(checkSum = sum(!!sym("childVal") * !!sym("factor")), .groups = "drop") %>%
+      summarise(checkSum = sum(!!sym("childVal") * !!sym("factor"), na.rm = TRUE),
+        summation = paste(
+          ifelse(
+            is.na(!!sym("child")),
```

Comment:
Under which circumstances could the column "child" have NA values?

Reply:
The parent variable or the child variable? In the former case, you should not get an entry at all. In the latter case, you get NA in `childValue`, but not in `child`. Or am I missing something?

Reply:
Your are right. This was a necessary check when I was calculating the details column without enforcing the full list of parents and childs combinations at lines 
https://github.com/pik-piam/piamInterfaces/pull/186/files#:~:text=%23%20add%20NA,%3D%20TRUE)

fixed at https://github.com/pik-piam/piamInterfaces/pull/186/commits/7a5dd09908bb44337f451844497d6282e29fa2c7